src/EditorFeatures/Core.Cocoa/NavigationCommandHandlers/FindBaseSymbolsCommandHandler.cs

```diff
@@ -66,31 +66,31 @@ protected override bool TryExecuteCommand(int caretPosition, Document document,
             {
                 using var token = _asyncListener.BeginAsyncOperation(nameof(StreamingFindBaseSymbolsAsync));
 
-                var context = presenter.StartSearch(EditorFeaturesResources.Navigating, supportsReferences: true, cancellationToken);
+                (var context, cancellationToken) = presenter.StartSearch(EditorFeaturesResources.Navigating, supportsReferences: true, cancellationToken);
```

Comment:
So this pattern at first to me looked like an outright bug -- the fact we're now combining with a new cancellation token meant that this method could throw cancellation which was _not_ triggered by the original requestor, which is always a no-no. But it's even funkier: the caller is passing in CancellationToken.None!

Note this is not a bug being introduced by this PR -- this was previously broken, the mistake is now much more obvious. The fix would be to change this pattern a bit: the call to presenter.StartSearch should be moved back into TryExecuteCommand, and you can pass the cancellation token from the CommandExecutionContext. From there, Task.Run(StreamingFindBaseSymbolsAsync), and there you can pass the cancellation token you receive from StartSearch -- essentially think of it as:

1. During the initial starting of the search, any delay is only cancellable by the threaded wait dialog, since we don't have another UI affordance for the user to do anything else.
2. Once the presenter is up, there's a cancel button there, and that's what the user uses.

Reply:
(note this comment applies to the other command handlers in this folder, since the same pattern/problem applies to all)

Reply:
yup.  this has been fixed.